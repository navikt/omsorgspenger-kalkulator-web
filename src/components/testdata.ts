import Barn from '../types/Barn';

export const treBarnEttKroniskOgAleneomsorg: Barn[] = [
  {
    s√∏kerHarAleneomsorgFor: true,
    kroniskSykt: true,
    alder: 'under12',
    id: '1',
  },
  {
    alder: 'under12',
    id: '2',
  },
  {
    alder: 'under12',
    id: '3',
  },
];
