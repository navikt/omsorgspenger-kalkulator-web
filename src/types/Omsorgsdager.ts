interface Omsorgsdager {
  normaldager: number;
  koronadager: number;
}

export default Omsorgsdager;
