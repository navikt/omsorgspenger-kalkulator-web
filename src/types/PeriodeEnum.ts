enum PeriodeEnum {
  Koronaperiode = 'Koronaperiode',
  UtenomKoronaperiode = 'UtenomKoronaperiode',
}

export default PeriodeEnum;
