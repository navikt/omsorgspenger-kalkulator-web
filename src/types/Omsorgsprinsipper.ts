import Omsorgsdager from './Omsorgsdager';

interface Omsorgsprinsipper {
  grunnrett: Omsorgsdager;
  kroniskSykt: Omsorgsdager;
  aleneomsorgKroniskSyke: Omsorgsdager;
  aleneomsorg: Omsorgsdager;
}

export default Omsorgsprinsipper;
