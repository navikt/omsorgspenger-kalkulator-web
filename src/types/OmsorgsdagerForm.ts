import Barn from './Barn';
import Forelder from './Forelder';
import PeriodeEnum from './PeriodeEnum';

interface OmsorgsdagerForm {
  periode?: PeriodeEnum;
  barn: Barn[];
  foreldre: Forelder[];
}

export default OmsorgsdagerForm;
