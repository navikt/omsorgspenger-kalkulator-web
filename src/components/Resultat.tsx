import React, { FunctionComponent, useMemo } from 'react';
import { useFormikContext } from 'formik';
import { Element } from 'nav-frontend-typografi';
import 'nav-frontend-tabell-style';
import { omsorgsdager } from './kalkulerOmsorgsdager';
import OmsorgsdagerForm from '../types/OmsorgsdagerForm';
import Omsorgsprinsipper from '../types/Omsorgsprinsipper';
import tekster from '../tekster';

export const summerDager = (omsorgsprinsipper: Omsorgsprinsipper): number => {
  return Object.values(omsorgsprinsipper)
    .filter(dag => !!dag)
    .reduce((sum, omsorgsdag) => sum + omsorgsdag.normaldager + omsorgsdag.koronadager, 0);
};

const Resultat: FunctionComponent = () => {
  const { values } = useFormikContext<OmsorgsdagerForm>();
  const totaltAntallDager = useMemo(() => omsorgsdager(values), [values]);

  return (
    <>
      <div className="arrowUp" />
      <div className="resultatHeader">
        <Element>{tekster('SøkerInput.HarRettPå')}</Element>
        <Element className="dagerOmsorg">
          {totaltAntallDager ? `${summerDager(totaltAntallDager)} ${tekster('SøkerInput.DagerOmsorgspenger')}` : '-'}
        </Element>
      </div>
      <table className="tabell tabell--stripet">
        <thead>
          <tr>
            <th />
            <th>{tekster('SøkerInput.Dager')}</th>
            <th>{tekster('SøkerInput.KoronaTillegg')}</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>{tekster('SøkerInput.Grunnrett')}</td>
            <td>{totaltAntallDager?.grunnrett.normaldager || '-'}</td>
            <td className="koronabakgrunn">{totaltAntallDager?.grunnrett.koronadager || '-'}</td>
          </tr>
          <tr>
            <td>{tekster('SøkerInput.KroniskSykdom')}</td>
            <td>{totaltAntallDager?.kroniskSykt.normaldager || '-'}</td>
            <td className="koronabakgrunn">{totaltAntallDager?.kroniskSykt.koronadager || '-'}</td>
          </tr>
          <tr>
            <td>{tekster('SøkerInput.AleneKroniskSykdom')}</td>
            <td>{totaltAntallDager?.aleneomsorgKroniskSyke.normaldager || '-'}</td>
            <td className="koronabakgrunn">{totaltAntallDager?.aleneomsorgKroniskSyke.koronadager || '-'}</td>
          </tr>
          <tr>
            <td>{tekster('SøkerInput.AleneOmOmsorg')}</td>
            <td>{totaltAntallDager?.aleneomsorg.normaldager || '-'}</td>
            <td className="koronabakgrunn">{totaltAntallDager?.aleneomsorg.koronadager || '-'}</td>
          </tr>
          <tr>
            <td>{tekster('SøkerInput.OverførtMottatt')}</td>
            <td>{totaltAntallDager?.overføringsdager.normaldager || '-'}</td>
            <td className="koronabakgrunn">{totaltAntallDager?.overføringsdager.koronadager || '-'}</td>
          </tr>
        </tbody>
      </table>
    </>
  );
};

export default Resultat;
