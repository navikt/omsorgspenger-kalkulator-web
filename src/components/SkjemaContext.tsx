import React from 'react';
import { Formik } from 'formik';
import OmsorgsdagerForm from '../types/OmsorgsdagerForm';

interface SkjemaProps {
  initialValues: OmsorgsdagerForm;
}

const noop = () => {};

const SkjemaContext: React.FunctionComponent<SkjemaProps> = ({ initialValues, children }) => {
  return (
    <Formik initialValues={initialValues} onSubmit={noop}>
      {children}
    </Formik>
  );
};

export default SkjemaContext;
