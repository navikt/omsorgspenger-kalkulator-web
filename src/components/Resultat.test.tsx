import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import Resultat, { summerDager } from './Resultat';
import SkjemaContext from './SkjemaContext';
import { barnUnder12, treBarnEttKroniskOgAleneomsorg } from './testdata';
import Omsorgsprinsipper from '@navikt/kalkuler-omsorgsdager/lib/types/Omsorgsprinsipper';
import Omsorgsdager from '@navikt/kalkuler-omsorgsdager/lib/types/Omsorgsdager';
import OmsorgsdagerForm from '@navikt/kalkuler-omsorgsdager/lib/types/OmsorgsdagerForm';
import PeriodeEnum from '@navikt/kalkuler-omsorgsdager/lib/types/PeriodeEnum';

test('Summerer alle omsorgsprinsipper', () => {
  const omsorgsprinsipper: Omsorgsprinsipper = {
    grunnrett: {
      normaldager: 15,
      koronadager: 15,
    },
    kroniskSykt: {
      normaldager: 10,
      koronadager: 10,
    },
    aleneomsorg: {
      normaldager: 10,
      koronadager: 10,
    },
    aleneomsorgKroniskSyke: {
      normaldager: 20,
      koronadager: 20,
    },
  };

  const overføringsdager: Omsorgsdager = {
    koronadager: -12,
    normaldager: -4,
  };
  const sum = summerDager(...Object.values(omsorgsprinsipper), overføringsdager);

  expect(sum).toEqual(94);
});

test('Rendrer riktig resultat', () => {
  const initValues: OmsorgsdagerForm = {
    barn: treBarnEttKroniskOgAleneomsorg,
    foreldre: [],
    periode: PeriodeEnum.Koronaperiode,
  };

  const { asFragment } = render(
    <SkjemaContext initialValues={initValues}>
      <Resultat />
    </SkjemaContext>,
  );

  expect(asFragment()).toMatchSnapshot();
});

test('Rendrer advarsel hvis man overfører fler dager enn man kan', () => {
  const initValues: OmsorgsdagerForm = {
    barn: [barnUnder12],
    foreldre: [
      {
        id: '1',
        normaldager: {
          dagerFått: 0,
          dagerTildelt: 11,
        },
        koronadager: {
          dagerTildelt: 11,
          dagerFått: 0,
        },
      },
    ],
    periode: PeriodeEnum.Koronaperiode,
  };

  const { asFragment } = render(
    <SkjemaContext initialValues={initValues}>
      <Resultat />
    </SkjemaContext>,
  );

  expect(asFragment()).toMatchSnapshot();
});

describe('Visning av Resultat med/uten koronafelter', () => {
  const rendered = (periode: PeriodeEnum): RenderResult => {
    const contextValues: OmsorgsdagerForm = {
      barn: [barnUnder12],
      foreldre: [
        {
          id: '1',
          normaldager: {
            dagerFått: 0,
            dagerTildelt: 11,
          },
          koronadager: {
            dagerTildelt: 11,
            dagerFått: 0,
          },
        },
      ],
      periode,
    };

    return render(
      <SkjemaContext initialValues={contextValues}>
        <Resultat />
      </SkjemaContext>,
    );
  };

  test('Periode innen korona viser koronafelter', () => {
    const { getByText } = rendered(PeriodeEnum.Koronaperiode);
    expect(getByText('Korona-tillegg')).toBeDefined();
  });

  test('Periode utenfor korona viser ikke koronafelter', () => {
    const { queryByText } = rendered(PeriodeEnum.UtenomKoronaperiode);
    expect(queryByText('Korona-tillegg')).toBeNull();
  });
});
