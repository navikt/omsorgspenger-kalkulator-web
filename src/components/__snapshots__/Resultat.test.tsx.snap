// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendrer riktig resultat 1`] = `
<DocumentFragment>
  <div
    class="resultatContainer"
  >
    <div
      class="resultatHeader"
    >
      <p
        class="typo-element"
      >
        Søkeren har rett på
      </p>
      <p
        class="typo-element dagerOmsorg"
      >
        90 dager med omsorgspenger
      </p>
    </div>
    <table
      class="tabell tabell--stripet"
    >
      <thead>
        <tr>
          <th />
          <th>
            Dager
          </th>
          <th>
            Korona-tillegg
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Grunnrett
          </td>
          <td>
            15
          </td>
          <td
            class="koronabakgrunn"
          >
            15
          </td>
        </tr>
        <tr>
          <td>
            Barn med kronisk sykdom
          </td>
          <td>
            -
          </td>
          <td
            class="koronabakgrunn"
          >
            -
          </td>
        </tr>
        <tr>
          <td>
            Aleneomsorg for barn med kronisk sykdom
          </td>
          <td>
            20
          </td>
          <td
            class="koronabakgrunn"
          >
            20
          </td>
        </tr>
        <tr>
          <td>
            Alene om omsorgen
          </td>
          <td>
            10
          </td>
          <td
            class="koronabakgrunn"
          >
            10
          </td>
        </tr>
        <tr>
          <td>
            Overført/Mottatt
          </td>
          <td>
            -
          </td>
          <td
            class="koronabakgrunn"
          >
            -
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
